processing/src/main/java/org/apache/druid/math/expr/FunctionalExpr.java

```diff
@@ -186,7 +187,15 @@ public String toString()
   @Override
   public ExprEval eval(ObjectBinding bindings)
   {
-    return function.apply(args, bindings);
+    try {
+      return function.apply(args, bindings);
+    }
+    catch (DruidException | ExpressionValidationException e) {
+      throw e;
+    }
+    catch (Exception e) {
+      throw DruidException.defensive().build(e, "Invocation of function '%s' encountered exception.", name);
```

Comment:
I think some other category would make more sense than "defensive". Defensive is sort of like an assert (it signals that we believe this should never happen unless there's a bug) but we do in fact expect functions to throw exceptions under "normal" conditions sometimes. (Like if they receive arguments of the wrong type, etc)

Reply:
In case the invoked function have already thrown out some reasonable exception (`DruidException`) - that's not touched at all - because a catch above caught it.

I wanted to add this to catch cases in which some unexpected issue happened while the function was processing....so that we don't let out a `NullPointerException` ; instead say that "Invocation of function x encountered ... " - it may give some help to the user what might have been the problem ; and it could give better error message to support instead of just refering to it as an NPE.

or do you think it would be better to repackage also normal `DruidExcpetion`-s and re-packaged? - and rethrown them with a bit more context?
so it will be something like:
```
DruidException: while invoking function x
caused by:
DruidException: Can't process an empty set because it Sunday!
``

Reply:
Ah I see. I thought this was wrapping all exceptions, but misread it. 

In the case I'm wondering if the "user-caused" exceptions (bad input) thrown by functions are all proper DruidExceptions? In that case a defensive might make sense here. We just want to avoid defensive for anything that isn't a code bug on our end.

Reply:
yes - other exceptions are good; the case was like: an unexpected issue happened - and the NPE bubbled up to the top. Giving little-to-no detail to the user about what happened; as the message didn't contained anything usefull.

I think if the user could see a bit friendlier message - it might give a starting point to either start narrowing it down the issue; or try to work it around